src/olympia/versions/models.py

```diff
@@ -728,18 +728,22 @@ def get_background_images_encoded(self, header_only=False):
                 file_obj, theme_data=None, header_only=header_only).items()}
 
     def can_be_disabled_and_deleted(self):
+        # see https://github.com/mozilla/addons-server/issues/15121#issuecomment-667226959  # noqa
+        # "It should apply to the <groups> that require a review to be badged"
         from olympia.promoted.models import PromotedApproval
 
-        if (self != self.addon.current_version or
-                not self.addon.promoted_group(group=RECOMMENDED)):
+        if self != self.addon.current_version or (
+            not (group := self.addon.promoted_group()) or
```

Comment:
Won't this condition always equate to `True`? Isn't this saying "if the group that is currently returned by `self.addon.promoted_group()` is `self.addon.promoted_group()`"? If this is the case, it seems overly complex to code it this way, rather than simply assigning `group = self.addon.promoted_group()` above the `if` and removing this clause from the `if`.

Reply:
> Won't this condition always equate to True?

how so?

> Isn't this saying "if the group that is currently returned by self.addon.promoted_group() is self.addon.promoted_group()"? 

it's assigning `group` to the value of `self.addon.promoted_group()` and then testing the group is truthy (`not` truthy in this case)

> If this is the case, it seems overly complex to code it this way, rather than simply assigning group = self.addon.promoted_group() above the if and removing this clause from the if.

we don't want to run the database query in `Addon.promoted_group` if it's not the current version of the add-on.  There are certainly other implementations, but this seemed like the ideal use for the new (for 3.8) operator.

Reply:
I misinterpreted the code. I thought it was both assigning and testing. It's both assigning and returning.